--- theme.inc
+++ fix.theme.inc
@@ -230,10 +230,12 @@
   if (!empty($items)) {
     $output .= "<$type" . drupal_attributes($attributes) . '>';
     $num_items = count($items);
-    foreach ($items as $i => $item) {
+    $i = 0;
+    foreach ($items as $item) {
       $attributes = array();
       $children = array();
       $data = '';
+      $i++;
       if (is_array($item)) {
         foreach ($item as $key => $value) {
           if ($key == 'data') {
@@ -263,10 +265,10 @@
         else {
           $attributes['class'][] = 'even';
         }
-        if ($i == 0) {
+        if ($i == 1) {
           $attributes['class'][] = 'first';
         }
-        if ($i == $num_items - 1) {
+        if ($i == $num_items) {
           $attributes['class'][] = 'last';
         }
       }

